packages/teams-js/test/private/files.spec.ts

```diff
@@ -576,15 +576,36 @@ describe('files', () => {
     it('should send the message to parent correctly', () => {
       utils.initializeWithContext('content');
 
-      const callback = jest.fn(err => {
+      const callback = jest.fn((err, provider) => {
         expect(err).toBeFalsy();
+        expect(provider).toEqual(files.CloudStorageProvider.Dropbox);
+      });
+
+      files.addCloudStorageProvider(callback);
+
+      const addCloudStorageProviderMessage = utils.findMessageByFunc('files.addCloudStorageProvider');
+      expect(addCloudStorageProviderMessage).not.toBeNull();
+      utils.respondToMessage(addCloudStorageProviderMessage, false, files.CloudStorageProvider.Dropbox);
+      expect(callback).toHaveBeenCalled();
+    });
+
+    it('should send the message to parent correctly and handle error scenario', () => {
+      utils.initializeWithContext('content');
+
+      const sdkError: SdkError = {
+        errorCode: ErrorCode.INTERNAL_ERROR,
+        message: 'Error Message',
+      };
+
+      const callback = jest.fn(err => {
+        expect(err).toEqual(sdkError);
       });
 
       files.addCloudStorageProvider(callback);
 
       const addCloudStorageProviderMessage = utils.findMessageByFunc('files.addCloudStorageProvider');
       expect(addCloudStorageProviderMessage).not.toBeNull();
-      utils.respondToMessage(addCloudStorageProviderMessage, false);
+      utils.respondToMessage(addCloudStorageProviderMessage, sdkError);
       expect(callback).toHaveBeenCalled();
```

Comment:
Can you add verification that the `provider` is not sent back in the error case?

Reply:
Added the verification.